core/common/src/main/java/alluxio/file/SeekableBufferedInputStream.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.file;
+
+import alluxio.Seekable;
+
+import com.google.common.base.Preconditions;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * A class optimizes seekable input streams by providing buffering.
+ */
+public class SeekableBufferedInputStream extends BufferedInputStream implements Seekable {
+
+  /**
+   * Constructor.
+   * @param in input stream implements seekable
+   * @param size the buffer size
+   */
+  public SeekableBufferedInputStream(InputStream in, int size) {
+    super(in, size);
+    Preconditions.checkArgument(in instanceof Seekable,
```

Comment:
nit: check before super

Reply:
in constructor, they require that super() in the first line before all other logics (including preconditions.check)

I was also confusing why this is enforced that way:
https://stackoverflow.com/questions/1168345/why-do-this-and-super-have-to-be-the-first-statement-in-a-constructor